src/serializer/utils.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { ObjectValue } from "../values/index.js";
+import type { Realm } from "../realm.js";
+
+import invariant from "../invariant.js";
+import { IsArray } from "../methods/index.js";
+
+/**
+ * Get index property list length by searching array properties list for the max index key value plus 1.
+ */
+export function getArrayIndexPropertyLength(realm: Realm, val: ObjectValue): number {
+  invariant(IsArray(realm, val));
+
+  let length = 0;
+  for (const key of val.properties.keys()) {
+    const isArrayIndex = String(Number(key)) === key;
```

Comment:
This consider values that are not actually valid array indices, e.g. `Math.pow(2, 64)`.
There is already a function that checks that: `IsArrayIndex`.